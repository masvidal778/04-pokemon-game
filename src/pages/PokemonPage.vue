<script>
import {defineComponent} from 'vue'
import PokemonOptions from "@/components/PokemonOptions.vue";
import PokemonPicture from "@/components/PokemonPicture.vue";

export default defineComponent({
  name: "PokemonPage",
  components: {
    PokemonPicture,
    PokemonOptions
  }
})
</script>

<template>

  <h1>Qui Ã©s aquest Pokemon?</h1>

  <PokemonPicture />
  <PokemonOptions />

</template>