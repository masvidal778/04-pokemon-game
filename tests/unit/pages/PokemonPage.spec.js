import { shallowMount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage.vue";
import { mockPokemons } from "../mocks/pokemons.mock";

describe('PokemonPage Component', () => {

    let wrapper

    beforeEach( () => {

        wrapper = shallowMount( PokemonPage )

    })

    test('ha de coincidir amb snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    })

    test('ha de cridar mixPokemonArray al muntar', () => {

        const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        const wrapper = shallowMount( PokemonPage )

        expect(mixPokemonArraySpy).toHaveBeenCalled()
    })

    test('ha de fer match amb el snapshot quan carreguen els pokemons', () => {

        const wrapper = shallowMount( PokemonPage, {
            data() {
                return {
                    pokemonArr: mockPokemons,
                    pokemon: mockPokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        expect( wrapper.html() ).toMatchSnapshot()

    })

    test('ha de mostrar els components de PokemonPicture i PokemonOptions', () => {

        const wrapper = shallowMount( PokemonPage, {
            data() {
                return {
                    pokemonArr: mockPokemons,
                    pokemon: mockPokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        const pictureWrapper = wrapper.find('pokemon-picture-stub')
        const optionsWrapper = wrapper.find('pokemon-options-stub')

        expect(pictureWrapper.exists()).toBeTruthy()
        expect(optionsWrapper.exists()).toBeTruthy()

        expect(pictureWrapper.attributes('pokemonid')).toBe("5")
        expect(optionsWrapper.attributes('pokemons')).toBeTruthy()

    })

    test('proves amb checkAnswer', async() => {

        const wrapper = shallowMount( PokemonPage, {
            data() {
                return {
                    pokemonArr: mockPokemons,
                    pokemon: mockPokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        await wrapper.vm.checkAnswer(5)

        expect( wrapper.find('h2').exists() ).toBeTruthy()
        expect( wrapper.vm.showPokemon).toBe(true)
        expect( wrapper.find('h2').text() ).toBe(`Correcte! Ã‰s ${mockPokemons[0].name}`)

        await wrapper.vm.checkAnswer(10)

        expect(wrapper.vm.message).toBe(`Ooops! Era ${mockPokemons[0].name}`)
    })

})