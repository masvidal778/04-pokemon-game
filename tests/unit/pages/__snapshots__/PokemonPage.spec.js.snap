// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component ha de coincidir amb snapshot 1`] = `<h1>Esperi, sisplau...</h1>`;

exports[`PokemonPage Component ha de fer match amb el snapshot quan carreguen els pokemons 1`] = `
<div>
  <h1>Qui Ã©s aquest Pokemon?</h1>
  <pokemon-picture-stub pokemonid="5" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
